pkg/devicerepository/grpc.go

```diff
@@ -60,13 +61,31 @@ func (dr *DeviceRepository) ensureBaseAssetURLs(models []*ttnpb.EndDeviceModel)
 
 const defaultLimit = 1000
 
-func requireApplication(ctx context.Context, ids *ttnpb.ApplicationIdentifiers) error {
-	return rights.RequireApplication(ctx, ids, ttnpb.Right_RIGHT_APPLICATION_DEVICES_READ)
+var errNotAuthenticated = errors.DefineUnauthenticated(
+	"no_rights_in_context",
+	"Not authenticated, need to be logged in to make this request",
+)
+
+func requireAuthentication(ctx context.Context) error {
+	log.FromContext(ctx).Debug("Validating request authentication")
+	authInfo, err := rights.AuthInfo(ctx)
+	if err != nil {
+		log.FromContext(ctx).WithError(err).Debug("Error validating user's auth")
+		return err
+	}
+	if authInfo == nil {
+		log.FromContext(ctx).Debug("User not authenticated")
```

Comment:
`No authentication response available`. 

Also see my comment below on why I'd prefer not to use `nil` but instead a different check.